tests/api_test.py

```diff
@@ -69,3 +69,21 @@ def test_ignore_absolute_paths():
     assert get_path_if_relative(
         '/usr/lib/shared/schema.json',
     ) is None
+
+
+def test_resolve_nested_refs():
+    """
+    Make sure we resolve nested refs gracefully and not get lost in
+    the recursion. Also make sure we don't rely on dictionary order
+    """
+    import os
```

Comment:
Let's move these to the top of this python file instead of inlining them in our test